src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -1700,6 +1744,14 @@ fn resolution_failure(
                         item(res),
                         variant
                     ),
+                    ResolutionFailure::MalformedGenerics(variant) => match variant {
+                        MalformedGenericsVariant::UnbalancedAngleBrackets => {
+                            String::from("unbalanced angle brackets")
+                        }
+                        MalformedGenericsVariant::MissingType => {
+                            String::from("missing type for generic parameters")
+                        }
+                    },
```

Comment:
No need to nest this.
```suggestion
                    ResolutionFailure::MalformedGenerics(MalformedGenericsVariant::UnbalancedAngleBrackets) => String::from("unbalanced angle brackets"),
                    ResolutionFailure::MalformedGenerics(MalformedGenericsVariant::MissingType) => String::from("missing type for generic parameters"),
```

Reply:
I thought it would be a bit easier to read if all the malformed generics errors were grouped together. Thoughts?

Reply:
🤷 I don't have a strong preference either way but I do like shorter code. Up to you.

Reply:
There are a bunch more cases now, so I think I'll leave it as-is.